### ISSUE TITLE
Add TypeScript type definitions.

### ISSUE THREAD

**Issue by dkushner** [4Catalyzer/farce#84]
Went looking to use `found-relay` with a TypeScript/React project and was unable to do so as all of the examples depend on `farce` and its history management, which does not have TypeScript type definitions available (that I could find).

**Comment by taion** [comment]
I'd be happy to take a PR adding them.

Though note that you're not going to be directly dealing with the Farce API much in using Found.

**Comment by dkushner** [comment]
@taion: Yeah, I'd noticed as much from the documentation, which is why it was a bit frustrating. Really small piece of the puzzle but lack of types stopped me. I'll see if I can at least get a decent start point out of dts-gen and go from there.

**Comment by taion** [comment]
Why do the types or lack thereof matter if you're not directly interacting with the library?

**Comment by dkushner** [comment]
@taion Per the example project for `found-relay`, I need to be able to instantiate the history protocol instance and import several other components.

```
import HashProtocol from 'farce/lib/HashProtocol';
import queryMiddleware from 'farce/lib/queryMiddleware';
```
This causes TypeScript to complain, obviously, because it can't find type definitions for these imports.
